src/cmap/connect.ts

```diff
@@ -117,157 +122,112 @@ function performInitialHandshake(
+            );
+          }
+          provider.auth(authContext, err => {
+            if (err) {
+              if (err instanceof MongoError) {
+                err.addErrorLabel(MongoErrorLabel.HandshakeError);
+                if (needsRetryableWriteLabel(err, response.maxWireVersion)) {
+                  err.addErrorLabel(MongoErrorLabel.RetryableWriteError);
+                }
               }
+              return callback(err);
             }
-            return callback(err);
-          }
-          callback(undefined, conn);
-        });
-
-        return;
-      }
+            callback(undefined, conn);
+          });
 
-      callback(undefined, conn);
-    });
-  });
-}
+          return;
+        }
 
-export interface HandshakeDocument extends Document {
-  /**
-   * @deprecated Use hello instead
-   */
-  ismaster?: boolean;
-  hello?: boolean;
-  helloOk?: boolean;
-  client: ClientMetadata;
-  compression: string[];
-  saslSupportedMechs?: string;
-  loadBalanced?: boolean;
+        callback(undefined, conn);
+      });
+    },
+    err => {
+      return callback(err);
+    }
+  );
 }
 
 /**
  * @internal
  *
  * This function is only exposed for testing purposes.
  */
-export function prepareHandshakeDocument(
-  authContext: AuthContext,
-  callback: Callback<HandshakeDocument>
-) {
-  const options = authContext.options;
-  const compressors = options.compressors ? options.compressors : [];
-  const { serverApi } = authContext.connection;
-
-  const handshakeDoc: HandshakeDocument = {
-    [serverApi?.version ? 'hello' : LEGACY_HELLO_COMMAND]: 1,
-    helloOk: true,
-    client: options.metadata || makeClientMetadata(options),
-    compression: compressors
-  };
-
-  if (options.loadBalanced === true) {
-    handshakeDoc.loadBalanced = true;
-  }
-
-  const credentials = authContext.credentials;
-  if (credentials) {
-    if (credentials.mechanism === AuthMechanism.MONGODB_DEFAULT && credentials.username) {
-      handshakeDoc.saslSupportedMechs = `${credentials.source}.${credentials.username}`;
-
-      const provider = AUTH_PROVIDERS.get(AuthMechanism.MONGODB_SCRAM_SHA256);
-      if (!provider) {
-        // This auth mechanism is always present.
-        return callback(
-          new MongoInvalidArgumentError(
-            `No AuthProvider for ${AuthMechanism.MONGODB_SCRAM_SHA256} defined.`
-          )
-        );
-      }
-      return provider.prepare(handshakeDoc, authContext, callback);
-    }
-    const provider = AUTH_PROVIDERS.get(credentials.mechanism);
-    if (!provider) {
-      return callback(
-        new MongoInvalidArgumentError(`No AuthProvider for ${credentials.mechanism} defined.`)
-      );
-    }
-    return provider.prepare(handshakeDoc, authContext, callback);
-  }
-  callback(undefined, handshakeDoc);
+export async function prepareHandshakeDocument(
+  authContext: AuthContext
+): Promise<HandshakeDocument> {
+  const { options } = authContext;
+  const generator = options.monitorHandshake
```

Comment:
The Monitor doesn't have an auth context - so we set an option to tell connect not to include the auth decorator.